src/behavior.rs

```diff
@@ -22,17 +22,37 @@ pub enum EditAction {
     TabSelected,
 }
 
+/// The state of a tab, used to inform the rendering of the tab.
+#[derive(Clone, Debug, Default)]
+pub struct TabState {
+    pub active: bool,
+    pub is_being_dragged: bool,
+    pub closable: bool,
+}
+
 /// Trait defining how the [`super::Tree`] and its panes should be shown.
 pub trait Behavior<Pane> {
     /// Show a pane tile in the given [`egui::Ui`].
     ///
     /// You can make the pane draggable by returning [`UiResponse::DragStarted`]
     /// when the user drags some handle.
-    fn pane_ui(&mut self, _ui: &mut Ui, _tile_id: TileId, _pane: &mut Pane) -> UiResponse;
+    fn pane_ui(&mut self, ui: &mut Ui, tile_id: TileId, pane: &mut Pane) -> UiResponse;
 
     /// The title of a pane tab.
     fn tab_title_for_pane(&mut self, pane: &Pane) -> WidgetText;
 
+    /// Should the tab be closable?
+    fn closable(&self) -> bool {
+        false
+    }
+
+    /// If `closable()` returns true, hat should be done when the close button is pressed?
+    ///
+    /// Return `true` if the tab should be closed, `false` otherwise.
+    fn on_tab_close(&mut self, _tiles: &mut Tiles<Pane>, _tile_id: TileId) -> bool {
+        false
+    }
```

Comment:
We should consider removing this and instead have

```suggestion
    /// Should the tab have a close-button?
	fn is_tab_closable(&self, tiles: &Tiles<Pane>, tile_id: TileId) -> bool {
	    false
	}    
	
	/// Called after a tab has been closed.
	fn on_tab_close(&mut self, _tiles: &mut Tiles<Pane>, _tile_id: TileId) {
    }
```

That way the user has exact control over which tabs should have a close-button.

Reply:
Agreed on the `closable` signature change, I do require the bool returning from `on_tab_close`, for example an implementation might want to prompt the user to save changes (or cancel closing the panel). That's what gives the user to cancel the close request.